src/Razor/Microsoft.AspNetCore.Razor.Tools/src/RewriteCssCommand.cs

```diff
@@ -18,6 +19,10 @@ namespace Microsoft.AspNetCore.Razor.Tools
 {
     internal class RewriteCssCommand : CommandBase
     {
+        private const string DeepCombinatorText = "::deep";
+        private static Regex DeepCombinatorRegex = new Regex($@"^{DeepCombinatorText}\s*");
```

Comment:
These should use timeouts. @blowdart prefers we use an field to stash the timeout: https://github.com/dotnet/aspnetcore/blob/cc1f23c5f8afb7d2a00405f19811d2372c4fcec2/src/Middleware/Rewrite/src/ApacheModRewrite/RuleBuilder.cs#L21

Reply:
```suggestion
        private static readonly Regex DeepCombinatorRegex = new Regex($@"^{DeepCombinatorText}\s*");
```

Reply:
This is build-time code, and runs only against the developer’s own source code. Additionally its evaluation will be linear. 

Is a timeout still applicable in that case, in your view? I don’t mind adding it, it just seems distracting given what we know about the situation.

Reply:
It probably just makes our lives easier if these APIs auto-scanned. AFAIK, adding the timeout doesn't affect the runtime characteristics of the regex.

Reply:
OK, no objections. Added.